cmd/basecmd/environmentcommand.go

```diff
@@ -282,36 +299,78 @@ func (c *EnvCommandBase) ConnectionName() string {
 	return c.envName
 }
 
+// WrapeSystemOption sets various parameters of the
+// EnvironCommand wrapper.
+type WrapEnvOption func(*environCommandWrapper)
+
+// EnvSkipFlags instructs the wrapper to skip --e and
+// --environment flag definition.
+func EnvSkipFlags() WrapEnvOption {
+	return func(w *environCommandWrapper) {
+		w.setFlags = false
+	}
+}
+
+// EnvSkipDefault instructs the wrapper not to
+// use the default environment.
+func EnvSkipDefault() WrapEnvOption {
+	return func(w *environCommandWrapper) {
+		w.useDefaultEnvironment = false
+	}
+}
+
 // Wrap wraps the specified EnvironCommand, returning a Command
 // that proxies to each of the EnvironCommand methods.
-func Wrap(c EnvironCommand) cmd.Command {
-	return &environCommandWrapper{EnvironCommand: c}
+func Wrap(c EnvironCommand, options ...WrapEnvOption) cmd.Command {
```

Comment:
// Any provided options are applied to the wrapped command
// before it is returned.